src/components/modals/TeachSessionModal.tsx

```diff
@@ -155,6 +173,22 @@ class TeachModal extends React.Component<Props, ComponentState> {
                 replaceActivityIndex
             })
         }
+
+        // If we just added a sourceTrainDialog it's because we continued from an existing Train Dialog
+        // Or if the tags or description were changed before continuing
+        // Copy over the tags and description from it
+        if (!this.props.sourceTrainDialog && newProps.sourceTrainDialog
```

Comment:
Logic conditions a bit hard to follow and validate; suggest more liberal use of parens, or reworking logic if possible